### PR TITLE
Improve grammar and fix typos for some documents

### PR DESCRIPTION
**Pull request by ngohoangthang**
I just want to improve the document. During working on this, I have used https://languagetool.org/ to automatically check the typos and grammar issues. 

Please let me know if there's any change you think unsuitable.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ There are many ways to contribute to React Native, and many of them do not invol
 * Browse [Stack Overflow](https://stackoverflow.com/questions/tagged/react-native) and answer questions. This will help you get familiarized with common pitfalls or misunderstandings, which can be useful when contributing updates to the documentation.
 * Take a look at the [features requested](https://react-native.canny.io/feature-requests) by others in the community and consider opening a pull request if you see something you want to work on.
 
-Contributions are very welcome. If you think you need help planning your contribution, please hop into [#react-native](https://discord.gg/0ZcbPKXt5bZjGY5n) and let people know you're looking for a mentor.
+Contributions are very welcome. If you think you need help to plain your contribution, please hop into [#react-native](https://discord.gg/0ZcbPKXt5bZjGY5n) and let people know you're looking for a mentor.
```
Comment (hramos) [inline docs/contributing.md]: `to plain` is misspelled.

File: docs/contributing.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ You can learn more about handling issues in the [maintainer's guide](maintainers
 
 ## Our development process
 
-Some of the core team will be working directly on [GitHub](https://github.com/facebook/react-native). These changes will be public from the beginning. Other changesets will come via a bridge with Facebook's internal source control. This is a necessity as it allows engineers at Facebook outside of the core team to move fast and contribute from an environment they are comfortable in.
+Some core team will be working directly on [GitHub](https://github.com/facebook/react-native). These changes will be public from the beginning. Other changesets will come via a bridge with Facebook's internal source control. This is a necessity as it allows engineers at Facebook outside of the core team to move fast and contribute from an environment they are comfortable in.
```
Comment (hramos) [inline docs/contributing.md]: Let's keep the original wording.

File: docs/contributing.md [hunk 3]
```diff
@@ -88,7 +88,7 @@ If you would like to request a new feature or enhancement but are not yet thinki
 
 If you intend to change the public API, or make any non-trivial changes to the implementation, we recommend [filing an issue](https://github.com/facebook/react-native/issues/new?title=%5BProposal%5D) that includes `[Proposal]` in the title. This lets us reach an agreement on your proposal before you put significant effort into it. These types of issues should be rare. If you have been contributing to the project long enough, you will probably already have access to the [React Native Core Contributors](https://www.facebook.com/groups/reactnativeoss/) Facebook Group, where this sort of discussion is usually held.
 
-If you're only fixing a bug, it's fine to submit a pull request right away but we still recommend to file an issue detailing what you're fixing. This is helpful in case we don't accept that specific fix but want to keep track of the issue.
+If you're only fixing a bug, it's fine to submit a pull request right away, but we still recommend filing an issue detailing what you're fixing. This is helpful in case we don't accept that specific fix but want to keep track of the issue.
```
Comment (hramos) [inline docs/contributing.md]: 👍

File: docs/custom-webview-ios.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ title: Custom WebView
 
 While the built-in web view has a lot of features, it is not possible to handle every use-case in React Native. You can, however, extend the web view with native code without forking React Native or duplicating all the existing web view code.
 
-Before you do this, you should be familiar with the concepts in [native UI components](native-components-ios). You should also familiarise yourself with the [native code for web views](https://github.com/facebook/react-native/blob/master/React/Views/RCTWebViewManager.m), as you will have to use this as a reference when implementing new features—although a deep understanding is not required.
+Before you do this, you should be familiar with the concepts in [native UI components](native-components-ios). You should also familiarize yourself with the [native code for web views](https://github.com/facebook/react-native/blob/master/React/Views/RCTWebViewManager.m), as you will have to use this as a reference when implementing new features—although a deep understanding is not required.
```
Comment (hramos) [inline docs/custom-webview-ios.md]: 👍

File: docs/custom-webview-ios.md [hunk 5]
```diff
@@ -174,7 +174,7 @@ const RCTCustomWebView = requireNativeComponent(
 
 If you want to add custom props to your native component, you can use `nativeConfig.props` on the web view. For iOS, you should also set the `nativeConfig.viewManager` prop with your custom WebView ViewManager as in the example above.
 
-For events, the event handler must always be set to a function. This means it isn't safe to use the event handler directly from `this.props`, as the user might not have provided one. The standard approach is to create a event handler in your class, and then invoking the event handler given in `this.props` if it exists.
+For events, the event handler must always be set to a function. This means it isn't safe to use the event handler directly from `this.props`, as the user might not have provided one. The standard approach is to create an event handler in your class, and then invoking the event handler given in `this.props` if it exists.
```
Comment (hramos) [inline docs/custom-webview-ios.md]: 👍

File: docs/debugging.md [hunk 6]
```diff
@@ -227,7 +227,7 @@ Follow this guide to enable Stetho for Debug mode:
     }
 ```
 
-4. Open the project in Android Studio and resolve any dependency issues. The IDE should guide you through this steps after hovering your pointer over the red lines.
+4. Open the project in Android Studio and resolve any dependency issues. The IDE should guide you through these steps after hovering your pointer over the red lines.
```
Comment (hramos) [inline docs/debugging.md]: 👍

File: docs/easing.md [hunk 7]
```diff
@@ -191,7 +191,7 @@ static elastic(bounciness)
 
 A simple elastic interaction, similar to a spring oscillating back and forth.
 
-Default bounciness is 1, which overshoots a little bit once. 0 bounciness doesn't overshoot at all, and bounciness of N > 1 will overshoot about N times.
+Default bounciness is 1, which overshoots a little once. 0 bounciness doesn't overshoot at all, and bounciness of N > 1 will overshoot about N times.
```
Comment (hramos) [inline docs/easing.md]: I don't think the problem here is "bit". The original sentence is lacking a comma before "once". Adding that should help.

File: docs/flatlist.md [hunk 8]
```diff
@@ -93,7 +93,7 @@ This is a convenience wrapper around [`<VirtualizedList>`](virtualizedlist.md),
 * In order to constrain memory and enable smooth scrolling, content is rendered asynchronously offscreen. This means it's possible to scroll faster than the fill rate and momentarily see blank content. This is a tradeoff that can be adjusted to suit the needs of each application, and we are working on improving it behind the scenes.
 * By default, the list looks for a `key` prop on each item and uses that for the React key. Alternatively, you can provide a custom `keyExtractor` prop.
 
-Also inherits [ScrollView Props](scrollview.md#props), unless it is nested in another FlatList of same orientation.
+Also, inherits [ScrollView Props](scrollview.md#props), unless it is nested in another FlatList of same orientation.
```
Comment (hramos) [inline docs/flatlist.md]: I don't think the comma is needed here.

File: docs/gesture-responder-system.md [hunk 9]
```diff
@@ -18,7 +18,7 @@ These features make users more comfortable while using an app, because it allows
 
 ### TouchableHighlight and Touchable\*
 
-The responder system can be complicated to use. So we have provided an abstract `Touchable` implementation for things that should be "tappable". This uses the responder system and allows you to easily configure tap interactions declaratively. Use `TouchableHighlight` anywhere where you would use a button or link on web.
+The responder system can be complicated to use. So we have provided an abstract `Touchable` implementation for things that should be "tappable". This uses the responder system and allows you to easily configure tap interactions decoratively. Use `TouchableHighlight` anywhere where you would use a button or link on web.
```
Comment (hramos) [inline docs/gesture-responder-system.md]: Please keep the original wording here.

File: docs/handling-touches.md [hunk 10]
```diff
@@ -98,7 +98,7 @@ Which "Touchable" component you use will depend on what kind of feedback you wan
 
 * If you need to handle a tap gesture but you don't want any feedback to be displayed, use [**TouchableWithoutFeedback**](touchablewithoutfeedback.md).
 
-In some cases, you may want to detect when a user presses and holds a view for a set amount of time. These long presses can be handled by passing a function to the `onLongPress` props of any of the "Touchable" components.
+In some cases, you may want to detect when a user presses and holds a view for a set amount of time. These long presses can be handled by passing a function to the `onLongPress` props of any the "Touchable" components.
```
Comment (hramos) [inline docs/handling-touches.md]: Please keep the original wording.

File: docs/accessibilityinfo.md [hunk 11]
```diff
@@ -3,7 +3,7 @@ id: accessibilityinfo
 title: AccessibilityInfo
 ---
 
-Sometimes it's useful to know whether or not the device has a screen reader that is currently active. The `AccessibilityInfo` API is designed for this purpose. You can use it to query the current state of the screen reader as well as to register to be notified when the state of the screen reader changes.
+Sometimes it's useful to know whether the device has a screen reader that is currently active. The `AccessibilityInfo` API is designed for this purpose. You can use it to query the current state of the screen reader as well as to register to be notified when the state of the screen reader changes.
```
Comment (hramos) [inline docs/accessibilityinfo.md]: 👍

File: docs/android-building-from-source.md [hunk 12]
```diff
@@ -109,7 +109,7 @@ dependencies {
 
 #### 4. Making 3rd-party modules use your fork
 
-If you use 3rd-party React Native modules, you need to override their dependencies so that they don't bundle the pre-compiled library. Otherwise you'll get an error while compiling - `Error: more than one library with package name 'com.facebook.react'`.
+If you use 3rd-party React Native modules, you need to override their dependencies so that they don't bundle the pre-compiled library. Otherwise, you'll get an error while compiling - `Error: more than one library with package name 'com.facebook.react'`.
```
Comment (hramos) [inline docs/android-building-from-source.md]: 👍

File: docs/animations.md [hunk 13]
```diff
@@ -9,7 +9,7 @@ React Native provides two complementary animation systems: [`Animated`](animatio
 
 ## `Animated` API
 
-The [`Animated`](animated.md) API is designed to make it very easy to concisely express a wide variety of interesting animation and interaction patterns in a very performant way. `Animated` focuses on declarative relationships between inputs and outputs, with configurable transforms in between, and simple `start`/`stop` methods to control time-based animation execution.
+The [`Animated`](animated.md) API is designed to make it very easy to concisely express a wide variety of interesting animation and interaction patterns in a very efficient way. `Animated` focuses on declarative relationships between inputs and outputs, with configurable transforms in between, and simple `start`/`stop` methods to control time-based animation execution.
```
Comment (hramos) [inline docs/animations.md]: I'd replace "very performant" with just "performant" here.

File: docs/animations.md [hunk 14]
```diff
@@ -139,7 +139,7 @@ Animated.spring(a, {
 
 ### Interpolation
 
-Each property can be run through an interpolation first. An interpolation maps input ranges to output ranges, typically using a linear interpolation but also supports easing functions. By default, it will extrapolate the curve beyond the ranges given, but you can also have it clamp the output value.
+Each property can be run through an interpolation first. An interpolation maps input ranges to output ranges, typically using a linear interpolation but also supports easing functions. By default, it will extrapolate the curve beyond the ranges given, but you can also have it clamps the output value.
```
Comment (hramos) [inline docs/animations.md]: Keep the original.

File: docs/animations.md [hunk 15]
```diff
@@ -164,7 +164,7 @@ For example, you may want to think about your `Animated.Value` as going from 0 t
   }}
 ```
 
-[`interpolate()`](animated.md#interpolate) supports multiple range segments as well, which is handy for defining dead zones and other handy tricks. For example, to get an negation relationship at -300 that goes to 0 at -100, then back up to 1 at 0, and then back down to zero at 100 followed by a dead-zone that remains at 0 for everything beyond that, you could do:
+[`interpolate()`](animated.md#interpolate) supports multiple range segments as well, which is handy for defining dead zones and other handy tricks. For example, to get a negation relationship at -300 that goes to 0 at -100, then back up to 1 at 0, and then back down to zero at 100 followed by a dead-zone that remains at 0 for everything beyond that, you could do:
```
Comment (hramos) [inline docs/animations.md]: 👍

File: docs/app-extensions.md [hunk 16]
```diff
@@ -7,7 +7,7 @@ App extensions let you provide custom functionality and content outside of your
 
 ## Memory use in extensions
 
-As these extensions are loaded outside of the regular app sandbox, it's highly likely that several of these app extensions will be loaded simultaneously. As you might expect, these extensions have small memory usage limits. Keep these in mind when developing your app extensions. It's always highly recommended to test your application on an actual device, and more so when developing app extensions: too frequently, developers find that their extension works just fine in the iOS Simulator, only to get user reports that their extension is not loading on actual devices.
+As these extensions are loaded outside of the regular app sandbox, it's highly likely that several of these app extensions will be loaded simultaneously. As you might expect, these extensions have small memory usage limits. Keep these in mind when developing your app extensions. It's always highly recommended testing your application on an actual device, and more so when developing app extensions: too frequently, developers find that their extension works just fine in the iOS Simulator, only to get user reports that their extension is not loading on actual devices.
```
Comment (hramos) [inline docs/app-extensions.md]: What changed here?

File: docs/appregistry.md [hunk 17]
```diff
@@ -18,7 +18,7 @@ title: AppRegistry
 
 To "stop" an application when a view should be destroyed, call `AppRegistry.unmountApplicationComponentAtRootTag` with the tag that was passed into `runApplication`. These should always be used as a pair.
 
-`AppRegistry` should be `require`d early in the `require` sequence to make sure the JS execution environment is setup before other modules are `require`d.
+`AppRegistry` should be `require`d early in the `require` sequence to make sure the JS execution environment is setup before other modules are `required`.
```
Comment (hramos) [inline docs/appregistry.md]: Keep the original, `required` is not a keyword. Alternatively, use required without \`backticks\`.

File: docs/building-for-apple-tv.md [hunk 18]
```diff
@@ -147,7 +147,7 @@ var running_on_android_tv = Platform.isTV;
 
 <block class="android" />
 
-* _Access to touchable controls_: When running on Android TV the Android framework will automatically apply a directional navigation scheme based on relative position of focusable elements in your views. The `Touchable` mixin has code added to detect focus changes and use existing methods to style the components properly and initiate the proper actions when the view is selected using the TV remote, so `TouchableHighlight`, `TouchableOpacity` and `TouchableNativeFeedback` will "just work". In particular:
+* _Access to touchable controls_: When running on Android TV the Android framework will automatically apply a directional navigation scheme based on the relative position of focusable elements in your views. The `Touchable` mixin has code added to detect focus changes and use existing methods to style the components properly and initiate the proper actions when the view is selected using the TV remote, so `TouchableHighlight`, `TouchableOpacity` and `TouchableNativeFeedback` will "just work". In particular:
```
Comment (hramos) [inline docs/building-for-apple-tv.md]: What changed here?

File: docs/cameraroll.md [hunk 19]
```diff
@@ -36,7 +36,7 @@ On Android, the tag must be a local image or video URI, such as `"file:///sdcard
 
 On iOS, the tag can be any image URI (including local, remote asset-library and base64 data URIs) or a local video file URI (remote or data URIs are not supported for saving video at this time).
 
-If the tag has a file extension of .mov or .mp4, it will be inferred as a video. Otherwise it will be treated as a photo. To override the automatic choice, you can pass an optional `type` parameter that must be one of 'photo' or 'video'.
+If the tag has a file extension of .mov or .mp4, it will be inferred as a video. Otherwise, it will be treated as a photo. To override the automatic choice, you can pass an optional `type` parameter that must be one of 'photo' or 'video'.
```
Comment (hramos) [inline docs/cameraroll.md]: 👍

File: docs/communication-android.md [hunk 20]
```diff
@@ -82,7 +82,7 @@ It is fine to update properties anytime. However, updates have to be performed o
 
 There is no way to update only a few properties at a time. We suggest that you build it into your own wrapper instead.
 
-> **_Note:_** Currently, JS functions `componentWillReceiveProps` and `componentWillUpdateProps` of the top level RN component will not be called after a prop update. However, you can access the new props in `componentWillMount` function.
+> **_Note:_** Currently, JS functions `componentWillReceiveProps` and `componentWillUpdateProps` of the top-level RN component will not be called after a prop update. However, you can access the new props in `componentWillMount` function.
```
Comment (hramos) [inline docs/communication-android.md]: 👍

File: docs/communication-android.md [hunk 21]
```diff
@@ -112,6 +112,6 @@ Events are powerful, because they allow us to change React Native components wit
 
 ### Calling native functions from React Native (native modules)
 
-Native modules are Java classes that are available in JS. Typically one instance of each module is created per JS bridge. They can export arbitrary functions and constants to React Native. They have been covered in detail in [this article](native-modules-android.md).
+Native modules are Java classes that are available in JS. Typically, one instance of each module is created per JS bridge. They can export arbitrary functions and constants to React Native. They have been covered in detail in [this article](native-modules-android.md).
```
Comment (hramos) [inline docs/communication-android.md]: 👍

File: docs/communication-ios.md [hunk 22]
```diff
@@ -58,7 +58,7 @@ class ImageBrowserApp extends React.Component {
 AppRegistry.registerComponent('AwesomeProject', () => ImageBrowserApp);
 ```
 
-`RCTRootView` also provides a read-write property `appProperties`. After `appProperties` is set, the React Native app is re-rendered with new properties. The update is only performed when the new updated properties differ from the previous ones.
+`RCTRootView` also provides a read-write property `appProperties`. After `appProperties` is set, React Native app is re-rendered with new properties. The update is only performed when the new updated properties differ from the previous ones.
```
Comment (hramos) [inline docs/communication-ios.md]: Keep the original.

File: docs/communication-ios.md [hunk 23]
```diff
@@ -99,11 +99,11 @@ Events are powerful, because they allow us to change React Native components wit
 * Events share namespace, which means that you may encounter some name collisions. Collisions will not be detected statically, which makes them hard to debug.
 * If you use several instances of the same React Native component and you want to distinguish them from the perspective of your event, you'll likely need to introduce identifiers and pass them along with events (you can use the native view's `reactTag` as an identifier).
 
-The common pattern we use when embedding native in React Native is to make the native component's RCTViewManager a delegate for the views, sending events back to JavaScript via the bridge. This keeps related event calls in one place.
+The common pattern we use when embedding native in React Native is to make the native component's RCTViewManager a delegate for the views, sending events back to JavaScript via the bridge. These keeps related event calls in one place.
```
Comment (hramos) [inline docs/communication-ios.md]: Keep the original.

File: docs/communication-ios.md [hunk 24]
```diff
@@ -99,11 +99,11 @@ Events are powerful, because they allow us to change React Native components wit
...
 
 ### Calling native functions from React Native (native modules)
 
-Native modules are Objective-C classes that are available in JS. Typically one instance of each module is created per JS bridge. They can export arbitrary functions and constants to React Native. They have been covered in detail in [this article](native-modules-ios.md#content).
+Native modules are Objective-C classes that are available in JS. Typically, one instance of each module is created per JS bridge. They can export arbitrary functions and constants to React Native. They have been covered in detail in [this article](native-modules-ios.md#content).
```
Comment (hramos) [inline docs/communication-ios.md]: 👍

File: docs/communication-ios.md [hunk 25]
```diff
@@ -199,7 +199,7 @@ Let's look at an example.
 }
 ```
 
-In the example we have a `FlexibleSizeExampleView` view that holds a root view. We create the root view, initialize it and set the delegate. The delegate will handle size updates. Then, we set the root view's size flexibility to `RCTRootViewSizeFlexibilityHeight`, which means that `rootViewDidChangeIntrinsicSize:` method will be called every time the React Native content changes its height. Finally, we set the root view's width and position. Note that we set there height as well, but it has no effect as we made the height RN-dependent.
+In the example we have a `FlexibleSizeExampleView` view that holds a root view. We create the root view, initialize it and set the delegate. The delegate will handle size updates. Then, we set the root view's size flexibility to `RCTRootViewSizeFlexibilityHeight`, which means that `rootViewDidChangeIntrinsicSize:` method will be called every time the React Native content changes its height. Finally, we set the root view's width and position. Note that we set their height as well, but it has no effect as we made the height RN-dependent.
```
Comment (hramos) [inline docs/communication-ios.md]: What changed here?

File: docs/contributing.md [hunk 26]
```diff
@@ -3,7 +3,7 @@ id: contributing
 title: How to Contribute
 ---
 
-React Native is one of Facebook's first open source projects that is both under very active development and is also being used to ship code to everybody using Facebook's mobile apps. If you're interested in contributing to React Native, hopefully this document makes the process for contributing clear.
+React Native is one of Facebook's first open source projects that is both under very active development and is also being used to ship code to everybody use Facebook's mobile apps. If you're interested in contributing to React Native, hopefully this document makes the process for contributing clear.
```
Comment (hramos) [inline docs/contributing.md]: Keep original.

**Comment by cpojer** [comment]
This is a stale PR so I'm going to close it. If you plan to work on this again, please submit a new PR :)

**Pull request closed without merging**